### PR TITLE
Backport REF 73: Delivery execution state controller

### PR DESCRIPTION
**Pull request by wazelin**
Backports #455.

Required for [`taoNfer`](https://github.com/oat-sa/extension-tao-nfer/pull/193).

### CODE REVIEW

File: doc/rest.json [hunk 1]
```diff
@@ -110,6 +110,77 @@
...
+            "description": "Execution identifier, in URI format",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "state",
+            "in": "body",
+            "description": "State to set the execution to",
+            "required": true,
+            "schema": {
```
Comment (ktsirkou) [inline doc/rest.json:138]: Why is this param an object and not simple string? 
Are you considering further extensions and allowing update of other properties of a deliver

File: doc/rest.json [hunk 2]
```diff
@@ -110,6 +110,77 @@
...
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "tags": [
+          "executions"
+        ],
+        "parameters": [
+          {
+            "name": "uri",
```
Comment (ktsirkou) [inline doc/rest.json:127]: I would rename this to `deliveryExecution` to be consistent with the `unstop` endpoint. 
What do you think?

File: doc/rest.json (hunk 1 above)
Comment (wazelin) [reply doc/rest.json:138]: I don't really expect it to be extended, but I prefer keeping the public interfaces easily extendible.
If it was just a single `string` that'd be the end of it :)

File: doc/rest.json (hunk 2 above)
Comment (wazelin) [reply doc/rest.json:127]: [`tao_actions_CommonRestModule`](https://github.com/oat-sa/tao-core/blob/master/actions/class.CommonRestModule.php#L346) wants it that way. And IMO we should stick to it for all the current-gen endpoints in order to keep the actions, performed on an entity, defined in a common manner.

**Comment by wazelin** [comment]
Since [the original PR](https://github.com/oat-sa/extension-tao-delivery/pull/455) has been approved, I consider this backport to be fine as well.

**Pull request merged**